### PR TITLE
INFRA-2314: Also Push to ECR

### PR DESCRIPTION
**Pull request by mohit**
Add conditional setting to push to ECR when the `ECR_PUSH_ID` (`aws_access_key_id`) is set.

### CODE REVIEW

File: circleci/docker-publish [hunk 1]
```diff
@@ -26,11 +40,19 @@ REPO=$CIRCLE_PROJECT_REPONAME
...
 
-echo "Logging into DockerHub..."
-docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL"
-
 echo "Building docker image..."
 docker build -t $ORG/$REPO:$SHORT_SHA .
 
-echo "Pushing docker image..."
+echo "Logging into DockerHub..."
+docker login -u $DOCKER_USER -p $DOCKER_PASS --email="$DOCKER_EMAIL"
```
Comment (rgarcia) [inline circleci/docker-publish]: curious why the switch in order (logging in and pushing)?

Comment (mohit) [inline circleci/docker-publish]: The order change I made was to build the image first and then login with a final step of push. 

Build is common to both ecr and dockerhub so is done first. Maybe I should pull dockerhub into its own function

Comment (mohit) [inline circleci/docker-publish]: @rgarcia what do you think about fd6c2b2df05f5cd407d92ec5921dbdfe0f4405c0

Comment (rgarcia) [inline circleci/docker-publish]: 👍

**Pull request merged**